pkg/engine/renderer/enhancer_test.go

```diff
@@ -141,6 +141,86 @@ func TestEnhancerApply_noAdditionalMetadata(t *testing.T) {
 	}
 }
 
+func TestEnhancerApply_dependencyHash(t *testing.T) {
+
+	ss := statefulSet("statefulset", "default")
+
+	ss.Spec.Template.Spec.Volumes = append(ss.Spec.Template.Spec.Volumes, corev1.Volume{
+		Name: "configMap",
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: "configmap"},
+			},
+		},
+	})
+
+	tpls := map[string]string{
+		"statefulset": resourceAsString(ss),
+		"configmap":   resourceAsString(configMap("configmap", "default")),
+	}
+
+	meta := metadata()
+	meta.PlanUID = uuid.NewUUID()
+
+	e := &DefaultEnhancer{
+		Scheme:    utils.Scheme(),
+		Discovery: fake.CachedDiscoveryClient(),
+	}
+
+	objs, err := e.Apply(tpls, meta)
+	if err != nil {
+		t.Errorf("failed to apply template %s", err)
+	}
+
+	for _, o := range objs {
+		unstructMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(o)
+		if err != nil {
+			t.Errorf("failed to parse object to unstructured: %s", err)
+		}
+		if o.(metav1.Object).GetName() != "statefulset" {
```

Comment:
Probably slightly easier to read, if you would pick the statefulset from the beginning:
```
ss := funk.Find(objs, func(o runtime.Object) bool {
    return o.GetObjectKind().GroupVersionKind() == schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "StatefulSet"}
})
```
but then again 🤷